File: Problem1/SourceCode/histogramEqualization.py
#!/usr/bin/python3
# -*- utf-8 -*-
# Reference: http://blog.csdn.net/jnulzl/article/details/47171549

from matplotlib import pyplot as plt 
import cv2
import numpy as np 

image = cv2.imread("../fig/Fig1.jpg")
image = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
plt.figure()
plt.title("Original Image")
plt.imshow(image)

# Histogram
hist = cv2.calcHist([image], [0], None, [256], [0, 256])

plt.figure()
plt.title("Grayscale Histogram")
plt.xlabel("Bins")
plt.ylabel("Number of Pixels")
plt.plot(hist)
plt.xlim([0, 256])

# Histogram Equalization
histeq = cv2.equalizeHist(image)
plt.figure()
plt.title("Histogram Equalization")
plt.imshow(histeq)
plt.show()

File: Problem1/SourceCode/histogramEqualization.cpp
#include <opencv2/opencv.hpp>
#include <iostream>
#include <stdio.h>

using namespace cv;
using namespace std;

/** @function main **/
int main( int argc, char ** argv )
{
    Mat src, dst;

    // Load original image
    src = imread("../fig/Fig1.jpg", 0);

    // Histogram Equalization
    equalizeHist( src, dst );
    // Show the result
    namedWindow( "Source image", CV_WINDOW_AUTOSIZE );
    namedWindow( "Equalized Image", CV_WINDOW_AUTOSIZE );

    imshow("Source image", src);
    imshow("Equalized Image", dst);

    // Wait for exit
    waitKey(0);

    return 0;
}
